src/ImuSensor_TEST.cc

```diff
@@ -511,6 +516,174 @@ TEST(ImuSensor_TEST, CustomRpyParentFrame)
   EXPECT_EQ(defultOrientValue, sensor->Orientation());
 }
 
+sdf::ElementPtr generateSensor(std::string namedFrame)
+{
+  std::ostringstream stream;
+  stream
+    << "<?xml version='1.0'?>"
+    << "<sdf version='1.6'>"
+    << " <model name='m1'>"
+    << "  <link name='link1'>"
+    << "    <sensor name='imu_sensor' type='imu'>"
+    << "      <topic>imu_test</topic>"
+    << "      <update_rate>1</update_rate>"
+    << "      <imu>"
+    << "      <orientation_reference_frame>"
+    << "      <localization>"+namedFrame+"</localization>"
+    << "      </orientation_reference_frame>"
+    << "      </imu>"
+    << "      <always_on>1</always_on>"
+    << "      <visualize>true</visualize>"
+    << "    </sensor>"
+    << "  </link>"
+    << " </model>"
+    << "</sdf>";
+
+  sdf::SDFPtr sdfParsed(new sdf::SDF());
+  sdf::init(sdfParsed);
+  if (!sdf::readString(stream.str(), sdfParsed))
+  {
+    return sdf::ElementPtr();
+  }
+  else
+  {
+    return sdfParsed->Root()->GetElement("model")->GetElement("link")
+      ->GetElement("sensor");
+  }
+}
+
+//////////////////////////////////////////////////
+TEST(ImuSensor_TEST, NamedFrameOrientationReference)
+{
+  // Create a sensor manager
+  ignition::sensors::Manager mgr;
+
+  math::Quaterniond orientValue;
+
+  // A. Localization tag is set to ENU
+  // ---------------------------------
+  auto sensorENU = mgr.CreateSensor<ignition::sensors::ImuSensor>(
+      generateSensor("ENU"));
+  ASSERT_NE(nullptr, sensorENU);
+
+  // Case A.1 : Localization ref: ENU, World : ENU
+  // Sensor aligns with ENU, we're measuring wrt ENU
+  sensorENU->SetWorldFrameOrientation(math::Quaterniond(0, 0, 0),
+    sensors::WorldFrameEnumType::ENU);
+  sensorENU->Update(std::chrono::steady_clock::duration(
+    std::chrono::nanoseconds(10000000)));
+  orientValue = math::Quaterniond(math::Vector3d(0, 0, 0));
+  EXPECT_EQ(orientValue, sensorENU->Orientation());
```

Comment:
it would be easier to read if we define the world frame orientation and enum type as a variable as well. We could enclose each of these test cases in a `{}` block so the names can have narrow scoping and be reused

for example:

~~~
{
  const math::Quaterniond worldFrameOrientation(0, 0, 0);
  const sensors::WorldFrameEnumType worldRelativeTo =
    sensors::WorldFrameEnumType::ENU;
  const math::Quaterniond expectedSensorOrientation(0, 0, 0);
  sensorENU->SetWorldFrameOrientation(worldFrameOrientation, worldRelativeTo);
  sensorENU->Update(std::chrono::steady_clock::duration(
    std::chrono::nanoseconds(10000000)));
  EXPECT_EQ(expectedSensorOrientation, sensorENU->Orientation());
}
~~~

how does that sound?

Reply:
Good idea. I'll refactor the tests.

Reply:
Done